components/eamxx/src/share/field/field_utils_impl_colred.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef SCREAM_FIELD_UTILS_IMPL_COLRED_HPP
+#define SCREAM_FIELD_UTILS_IMPL_COLRED_HPP
+
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+#include "ekat/mpi/ekat_comm.hpp"
+#include "share/field/field.hpp"
+
+namespace scream {
+namespace impl {
+
+// Utility to compute the reduction of a field along its column dimension.
+// This is equivalent to einsum('i,i...k->...k', f1, f2); i is the column.
+// The layouts are such that:
+// - The first dimension is for the columns (COL)
+// - There can be only up to 3 dimensions
+
+template <typename ST>
+Field column_reduction(const Field &f1, const Field &f2, const ekat::Comm *co) {
+  using KT          = ekat::KokkosTypes<DefaultDevice>;
+  using RangePolicy = Kokkos::RangePolicy<Field::device_t::execution_space>;
+  using TeamPolicy  = Kokkos::TeamPolicy<Field::device_t::execution_space>;
+  using TeamMember  = typename TeamPolicy::member_type;
+  using ESU         = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+  using namespace ShortFieldTagsNames;
+
+  const auto &l1 = f1.get_header().get_identifier().get_layout();
+
+  EKAT_REQUIRE_MSG(l1.rank() == 1,
+                   "Error! First field f1 must be rank-1.\n"
+                   "The input has rank "
+                       << l1.rank() << ".\n");
+  EKAT_REQUIRE_MSG(l1.tags() == std::vector<FieldTag>({COL}),
+                   "Error! First field f1 must have a column dimension.\n"
+                   "The input f1 layout is "
+                       << l1.tags() << ".\n");
+
+  const auto &n2 = f2.get_header().get_identifier().name();
+  const auto &l2 = f2.get_header().get_identifier().get_layout();
+  const auto &u2 = f2.get_header().get_identifier().get_units();
+  const auto &g2 = f2.get_header().get_identifier().get_grid_name();
+
+  EKAT_REQUIRE_MSG(l2.rank() <= 3,
+                   "Error! Second field f2 must be at most rank-3.\n"
+                   "The input f2 rank is "
+                       << l2.rank() << ".\n");
+  EKAT_REQUIRE_MSG(l2.tags()[0] == COL,
+                   "Error! Second field f2 must have a column dimension.\n"
+                   "The input f2 layout is "
+                       << l2.tags() << ".\n");
```

Comment:
If this works, it is b/c of some coincidence. You are calling `operator<<` on a `std::vector<FieldTag>`. There is no official overload of op<< for vector. You could however just do `layout is: l2.to_string() << "\n"`.